editor.planx.uk/src/@planx/components/MultipleFileUpload/Modal.tsx

```diff
@@ -93,29 +102,54 @@ const SelectMultiple = (props: SelectMultipleProps) => {
     );
   };
 
+  useEffect(() => {
+    const updatedFileList: FileList = merge(fileList);
+    tags.forEach((tag) => {
+      (
+        Object.keys(updatedFileList) as Array<keyof typeof updatedFileList>
+      ).forEach((category) => {
+        const updatedUserFileIndex = updatedFileList[category].findIndex(
+          (fileType) => fileType.name === tag
+        );
+        if (updatedUserFileIndex > -1) {
+          updatedFileList[category][updatedUserFileIndex] = {
+            ...updatedFileList[category][updatedUserFileIndex],
+            slot: uploadedFile,
+          };
+          setFileList(updatedFileList);
+        }
+      });
+    });
+  }, [tags]);
```

Comment:
If doing this update on every `tags` change starts to feel slow, I think another option is to call this in the `onClose` prop of `Select` - but we'll need to make sure only a single `Select` is open at a time in the modal which I'm not sure we're enforcing for yet.